### ISSUE TITLE
Consider updating hints and error messages to use paragraphs

### ISSUE THREAD

**Issue by 36degrees** [alphagov/govuk-frontend#2083]
## What

Consider updating hints and error messages to use paragraphs rather than spans and divs.

https://github.com/alphagov/govuk-frontend/blob/d5a3fa8c5e61dbc7ca0f7a8429cc8e6716ee7c32/src/govuk/components/error-message/template.njk#L3-L6

https://github.com/alphagov/govuk-frontend/blob/d5a3fa8c5e61dbc7ca0f7a8429cc8e6716ee7c32/src/govuk/components/hint/template.njk#L1-L4

## Why

As raised by a user on x-gov Slack:

> the error message is in a <span> tag. Is there a reason why a `<span>` is used, rather than a `<div>` or `<p>`, for example? The reason I ask is because I’ve been doing some testing with JAWS and I’ve found that there’s no pause between reading the hint text and the error message. So JAWS reads “For example 12 11 2007 error colon the date your passport was issued must be in the past”. Using a `<div>` didn’t make a difference, but using a `<p>` resulted in a pause between the hint text and the error message, so potentially making it a little clearer.

It looks like hints used to be paragraphs, but [were changed to spans when the hints were moved inside the label](https://github.com/alphagov/govuk_elements/commit/5f631c113840bfd4cc3ed81de8fb311638457a92#diff-1968332cde003a71f281b5d33445bcb3a43d2ac1cee565f6bc0e0a0144cfb846) – presumably as [only phrasing content is permitted within a `label`](https://html.spec.whatwg.org/multipage/forms.html#the-label-element). Error messages, when they were added later, were also spans inside the label.

We recently [updated hints to use divs rather than spans](https://github.com/alphagov/govuk-frontend/commit/d92d3c8a990730213430b686d239fc7b278530a9) so that they can be used with flow content (for example multiple paragraph tags, or a list)

Since [we moved hints and error messages out of the labels](https://github.com/alphagov/govuk-frontend/pull/681), we should consider whether they should now by marked up as paragraphs instead.

## Who needs to know about this

Developers, designers

## Done when

- [x] Tested alternate form controls with error messages and hints marked up as paragraphs in assistive technologies
- [x] Identified other potential issues cause by changing hints and error messages to paragraphs
- [ ] Decision made and code change made
- [ ] Check if the changelog for https://github.com/alphagov/govuk-frontend/issues/1856 needs updating as a result of this work

**Comment by 36degrees** [comment]
One consideration – we often see very long hints that are made up of multiple paragraphs, or include lists – so we might need to either keep hints as divs or make them configurable to support either a single paragraph or a div wrapper.

However, there are other issues with very long hints – they're associated with the input using `aria-describedby` which only really makes sense for shorter hints.

So, another alternative would be to be more restrictive about the sort of content that can go into a hint and provide an alternative pattern for cases where longer guidance is needed alongside a form input.

**Comment by vanitabarrett** [comment]
As discussed as part of our v4 prioritising, we should make sure:

- we test a variety of approaches with screenreaders to see which perform best
- we should make sure our implementation matches up with our guidance https://github.com/alphagov/govuk-design-system/pull/1716
- if we are considering moving to paragraphs, we need to think about the implications for services and what we recommend they do instead

We should remember that we can always tackle the error messages and hint text separately if we're more confident of one change than another.

**Comment by vanitabarrett** [comment]
Hints/Error messages

I’ve tested with Voiceover, NVDA and JAWS, comparing our current implementation (`<span>` for error messages and `<div>` for hints) with the proposed change (`<p>` for both). There doesn’t seem to be any difference in the way screen readers announce the hint or error message when it’s marked up as a paragraph.

➡️ [Full testing spreadsheet](https://docs.google.com/spreadsheets/d/1ARw5b_dXTuHdBC5GrbC7a19iwpxc6IKhOdvKrJtxXsQ/edit#gid=404092377  )

It feels pretty low risk to make this change for error messages. We know that hints are slightly more complex, as we did make a change previously from `<span>` to `<div>` to support more content which this slightly backtracks on. However, given the work we’ve done on https://github.com/alphagov/govuk-design-system/pull/1716 and the (I think) anecdotal evidence we’ve seen that long hints aren’t a great user experience for screen reader users, I think my vote would be to make both hints and error messages be marked up as paragraphs. We should be able to point people towards [the guidance for avoiding long hint text](https://design-system.service.gov.uk/patterns/question-pages/#asking-complex-questions-without-using-hint-text) in the release notes.

## Next Steps
If we agree that we want to use paragraphs for both hints and error messages, the next steps:

- Complete testing in Voiceover (iOS) and TalkBack (Android) to confirm everything works as expected
- Raise a PR to change the HTML templates to use paragraphs and update the tests accordingly
- Draft release notes, especially given guidance for users with long hint text

**Comment by vanitabarrett** [comment]
We've decided to go ahead and make the change for error messages. The change to hints needs a little more research as it'll block users from adding any block level elements to the hint. Generally we think this is ok as we already recommend not to have long hint text with lists or paragraphs. However, we want to explore what the actual effect of this in when using `aria-describedby` as we do, so we can call it out in the release notes if/when we make the change to hints.

**Comment by vanitabarrett** [comment]
This change has been made for error messages in https://github.com/alphagov/govuk-frontend/pull/2452, but I'm re-opening this issue as we haven't made the change for hints yet.

**Comment by edwardhorsford** [comment]
An example where a service might legitimately have a 'long hint', but isn't really long - where they have both a single line hint and an example for formatting.

A date might have a hint like:
```
This is the date when the event occurred
For example, 1 January 2021
```
